### PR TITLE
HDA 108

### PR DESCRIPTION
**Pull request by manthan-25**

### CODE REVIEW

File: src/BottomView.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  }
+
+  renderRemindLater = (onPress, laterButtonStyle) => (
+    <TouchableOpacity onPress={onPress}>
+      <Text style={[styles.later, laterButtonStyle]}>Later</Text>
+    </TouchableOpacity>
+  )
+
+  render() {
+    const { onButtonPress, onLaterPress, style, buttonType, buttonText } = this.props
```
Comment (VishwajeetDagar) [inline src/BottomView.js]: onPrimaryButtonPress

File: src/BottomView.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    const laterButtonStyle = (buttonType !== 'none') ? { marginLeft: 40 } : {}
+    return (
+      <View style={[styles.container, style]}>
+        {this.renderRemindLater(onLaterPress, laterButtonStyle)}
+        {(buttonType !== 'none') && this.renderButton(onButtonPress, buttonText, buttonType)}
```
Comment (VishwajeetDagar) [inline src/BottomView.js]: +1

File: src/BottomView.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    color: '#292929',
+    opacity: 0.5
+  }
+})
+
+export default class BottomView extends Component {
+
+  renderButton = (onPress, buttonText, buttonType) => {
+    const isDisabled = buttonType === 'disabled'
+    const color = isDisabled ? colors.black10 : '#1fd290'
```
Comment (VishwajeetDagar) [inline src/BottomView.js]: move to colors

File: src/FeedbackView.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import { View, TextInput, Text, StyleSheet, Image, TouchableOpacity } from 'react-native'
+import PropTypes from 'prop-types'
+import BottomView from './BottomView'
+import { isNilOrEmpty, closeImg } from './utils'
+
+const styles = StyleSheet.create({
+  title: {
+    fontWeight: '500',
+    fontSize: 18,
+    color: '#292929',
```
Comment (VishwajeetDagar) [inline src/FeedbackView.js]: move to colors

File: src/FeedbackView.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  },
+  bottomView: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    marginVertical: 35
+  }
+})
+
+const placeholder = 'E.g. I’m not able to access owner or agent contact details'
```
Comment (VishwajeetDagar) [inline src/FeedbackView.js]: use this as default prop, also extract from prop

File: src/FeedbackView.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        buttonText="Submit"
+        buttonType={buttonType}
+      />
+    )
+  }
+
+  render() {
+    const { onClose, onPressLater } = this.props
+    return (
+      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
```
Comment (VishwajeetDagar) [inline src/FeedbackView.js]: use scrollview

File: src/RatingComponent.js [hunk 7]
```diff
@@ -113,10 +82,30 @@ export default class RatingComponent extends Component {
...
-    this.props.sendEvent({ type: 'SUBMIT', ratingType, feedback })
+  onSubmit = (feedback = '') => {
+    const { rating } = this.state
+    const { thresholdRating, noOfDays, storeLink } = this.props
+    const isThresholdRating = rating >= thresholdRating
+    if( !isThresholdRating && (isNilOrEmpty(feedback) || typeof feedback !== 'string')) {
+      this.setState({ showFeedback: true })
+    }
+    else {
+      this.closeFeedback()
```
Comment (VishwajeetDagar) [inline src/RatingComponent.js]: move to different func

File: src/RatingComponent.js [hunk 8]
```diff
@@ -113,10 +82,30 @@ export default class RatingComponent extends Component {
...
+      const ratingType =  getRatingType(rating)
+      setShowDate(rating, noOfDays)
+      isThresholdRating && this.redirectToStore(storeLink)
+      this.showThankYouScreen()
+      this.props.sendEvent({ type: 'SUBMIT', ratingType, feedback })
+    }
+  }
+
+  closeFeedback = (callback = () => {}) => {
```
Comment (VishwajeetDagar) [inline src/RatingComponent.js]: change callback default value to null

**Review by VishwajeetDagar — APPROVED** [review]

**Review by AmandeepSingh06 — APPROVED** [review]

**Pull request merged**